doc/filters/merge.rst

```diff
@@ -1,7 +1,7 @@
 ``merge``
 =========
 
-The ``merge`` filter merges an array with another array:
+The ``merge`` filter merges an array or traversable with another array or traversable:
```

Comment:
I would not mention it here as we are talking about Twig, not PHP. Instead, I would add a note at the bottom explaining the PHP implementation.